src/Core/Validation/FieldValidation/FieldValidationTrait.php

```diff
@@ -21,13 +21,16 @@ trait FieldValidationTrait
      *
      * Each item in the array can be one of the following
      * a) MyFieldValidator::class,
-     * b) MyFieldValidator::class => [null, 'getSomething'],
-     * c) MyFieldValidator::class => null,
+     * b) MyFieldValidator::class => ['argNameA' => null, 'argNameB' => 'getSomething'],
+     * d) MyFieldValidator::class => null,
      *
      * a) Will create a MyFieldValidator and pass the name and value of the field as args to the constructor
      * b) Will create a MyFieldValidator and pass the name, value, and pass additional args, where each null values
      *    will be passed as null, and non-null values will call a method on the field e.g. will pass null for the first
      *    additional arg and call $field->getSomething() to get a value for the second additional arg
+     *    Keys are used to speicify the arg name, which is done to prevents duplicate
+     *    args being add to config when a subclass defines the same FieldValidator as a parent class.
+     *    Note that keys are not named args, they are simply arbitary keys
```

Comment:
That note is probably not useful (people using arbitrary keys would get confusing and could result in not overriding the correct values).
```suggestion
```

Alternatively if you really want this here, we should promote best practice
```suggestion
     *    Note that keys are not named args, they are simply arbitary keys - but best practice is for the keys to match constructor argument names.
```
(if it's not the constructor just put the relevant method name instead)

Reply:
Updated